Source/Particles/MultiParticleContainer.cpp

```diff
@@ -672,7 +672,7 @@ MultiParticleContainer::GetZeroParticlesInGrid (const int lev) const
 {
     const WarpX& warpx = WarpX::GetInstance();
     const int num_boxes = warpx.boxArray(lev).size();
-    const Vector<Long> r(num_boxes, 0);
+    Vector<Long> r(num_boxes, 0);
```

Comment:
This is exactly what I was referring to multiple weeks ago about const preventing moves and hurting performance.

Reply:
Yes! That's why I asked for your review! This check https://releases.llvm.org/14.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/performance-no-automatic-move.html should prevent this issue even if we enforce const correctness in the code.